### PR TITLE
fix(packages): implement proper NPI validation

### PR DESCRIPTION
**Pull request by Goncharo**
Ticket: metriport/metriport-internal#323

### Dependencies

N/A

### Description

Fix npi validation as per https://www.cms.gov/Regulations-and-Guidance/Administrative-Simplification/NationalProvIdentStand/Downloads/NPIcheckdigit.pdf

### Release Plan

ASAP

### CODE REVIEW

**Review by leite08 — CHANGES_REQUESTED** [review]
The logic looks good, considering we're not processing NPIs with prefix (80840) - which I assume is the case since we're changing the direction from rightmost to leftmost.

I just noticed we're not checking the first 2 digits - I might be missing some updates that would justify not checking those - in which case let's :shipit: . Otherwise, we might add that check as well.

**Comment by leite08** [comment]
And forgot to mention on the last comment: it would be great to have unit testing for this ;)

File: packages/packages/commonwell-sdk/src/common/validate-npi.ts [hunk 1]
```diff
@@ -10,20 +12,25 @@ export function validateNPI(npi: string): boolean {
   }
```
[the remark this answers is not in the document]
Comment (Goncharo) [reply packages/packages/commonwell-sdk/src/common/validate-npi.ts:12]: Confirming that we'll never be processing the prefix.

Good point about the first digit, will add that check as well.

File: packages/packages/commonwell-sdk/src/common/validate-npi.ts [hunk 2]
```diff
@@ -10,6 +10,10 @@ export function validateNPI(npi: string): boolean {
   if (npi.length !== 10) {
     return false;
   }
+  const firstNPIDigit = parseInt(npi.charAt(0), 10);
+  if (firstNPIDigit != 1 && firstNPIDigit != 2) {
```
Comment (leite08) [inline packages/packages/commonwell-sdk/src/common/validate-npi.ts:14]: !==

**Review by leite08 — APPROVED** [review]
:shipit:

**Pull request merged**